panther

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env php
+<?php
+
+/*
+ * This file is part of the Panther project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Psy\Shell;
+use Symfony\Component\Panther\Client;
+use Symfony\Component\Panther\PantherTestCaseTrait;
+
+require file_exists($autoload = __DIR__.'/vendor/autoload.php') ? $autoload : __DIR__.'/../../autoload.php';
+
+if (file_exists($psyshBin = $_SERVER['PANTHER_PSYSH_BIN'] ?? '/usr/local/bin/psysh')) {
+    require $psyshBin;
```

Comment:
requiring the phar file, especially by default (regardless of whether there's a local composer dependency installed) feels bad. it's just asking for version conflicts and incompatible third-party libraries. 

your whole session will use whatever version of Symfony components are shipped with the phar you happen to have on your system, rather than the ones that are depended on by your `composer.json`. or, if the local ones win, psysh won't have Symfony components that it expects. depending on how far the local phar version skews from what is in your composer lockfile, they might not be compatible at all, and we really have no way of knowing.

i think i would prefer always requiring a composer-installed `psy/psysh` dependency, rather than using the phar at all.

Reply:
if we _do_ support requiring the phar directly, we should respect the user's `$PATH` and load it that way. they might have a global composer install, a copy in `/usr/bin`, a per-user bin path in `~/bin`, etc. there's no way to tell, but it's almost certain we'll get it wrong for some users if we try to guess.

Reply:
```php
if (0 <= version_compare(Psy\Shell::version, 'v.0.9.10')) {
    echo 'Please upgrade your version of PsySH script.';
    die();
}
```
Will we need to do something like it ?

Reply:
>  it's almost certain we'll get it wrong for some users

To check it like below is probably safer:
```php
$psyshBin = strtok(exec(sprintf('%s psysh', '\\' === \DIRECTORY_SEPARATOR ? 'where' : 'command -v')), PHP_EOL)
```

Reply:
even if you require that version of psysh, you should still defer to a locally installed composer dependency rather than loading _anything_ from the phar.

Reply:
also, the scoped phar is a backwards compatibility break, so it will be at least v0.10.0.